app/models/redis_notification.rb

```diff
@@ -32,7 +37,18 @@ def self.track(action, subject)
     end
 
     if Setting.plugin_redmine_redis_notifier["enable_#{subject_type.underscore.pluralize}"] == "1" && notification_table_exists?
-      RedisNotification.create(action: action, subject_id: subject.id, subject_type: subject_type, current_user_id: User&.current&.id)
+      attributes = {
+        action: action,
+        subject_id: subject.id,
+        subject_type: subject_type,
+        current_user_id: User&.current&.id
+      }
+
+      ['project_id', 'issue_id', 'user_id', 'member_id', 'role_id'].each do |attribute|
+        attributes[attribute] = subject.send(attribute) if subject.has_attribute?(attribute)
+      end
+
+      RedisNotification.create(attributes)
```

Comment:
You may have already thought about it: If you prefer, you could replace `RedisNotification.create` by `create` (you're on class level here!) which would be more stable against class renamings or so. Otherwise... you may never rename the class. Was just thinking about it 😉